src/util/compareThought.ts

```diff
@@ -16,13 +16,31 @@ const IGNORED_PREFIXES = ['the ']
 const CURRENT_YEAR = new Date().getFullYear()
 
 const REGEX_PUNCTUATION = /^[!@#$%^&*()\-_=+[\]{};:'"<>.,?\\/].*/
-const REGEX_SHORT_DATE_WITH_DASH = /\d{1,2}-\d{1,2}/
-const REGEX_SHORT_DATE_WITH_SLASH = /\d{1,2}\/\d{1,2}/
 const REGEX_IGNORED_PREFIXES = new RegExp(`^(${IGNORED_PREFIXES.join('|')})(.*)`, 'gmi')
 const REGEX_FORMATTING = new RegExp(
   `^<(${ALLOWED_FORMATTING_TAGS.join('|')})[^>]*>(.*?)<\\s*/\\s*(${ALLOWED_FORMATTING_TAGS.join('|')})>`,
 )
 
+// Date pattern regex constants for performance optimization
+// Month names for reuse across date patterns (pipe-separated for regex)
+const MONTH_NAMES = 'January|February|March|April|May|June|July|August|September|October|November|December'
+
+// Combined regex for detecting all valid date formats:
+// - M/d or M/d/yyyy (slash format with optional year)
+// - M-d or M-d-yyyy (dash format with optional year)
+// - Month Day or Month Day, yyyy (written format with optional year)
+// Uses separate patterns to ensure consistent separators (no mixed slashes/dashes)
+const REGEX_DATE_COMBINED = new RegExp(
+  `^(\\d{1,2}\\/\\d{1,2}(\\/\\d{4})?|\\d{1,2}-\\d{1,2}(-\\d{4})?|(${MONTH_NAMES})\\s+\\d{1,2}(,\\s*\\d{4})?)$`,
+  'i',
+)
```

Comment:
Thanks, this looks better now. Ultimately I think we could have a single REGEX with an optional year, but we don't have to do that here.